koku/reporting/provider/gcp/models.py

```diff
@@ -949,3 +969,853 @@ class Meta:
+
+    source_uuid = models.ForeignKey(
+        "api.Provider", on_delete=models.CASCADE, unique=False, null=True, db_column="source_uuid"
+    )
+
+    invoice_month = models.CharField(max_length=256, null=True, blank=True)
+
+    credit_amount = models.DecimalField(max_digits=24, decimal_places=9, null=True, blank=True)
+
+
+class GCPComputeSummaryByProjectP(models.Model):
+    """A summarized partitioned table specifically for UI API queries.
+
+    This table gives a daily breakdown of total cost by account.
+
+    """
+
+    class PartitionInfo:
+        partition_type = "RANGE"
+        partition_cols = ["usage_start"]
+
+    class Meta:
+        """Meta for GCPComputeSummaryByProjectP."""
+
+        db_table = "reporting_gcp_compute_summary_by_project_p"
+        indexes = [
+            models.Index(fields=["usage_start"], name="gcpcompsumm_pro_usage_start"),
+            models.Index(fields=["instance_type"], name="gcpcompsumm_pro_insttyp"),
+            models.Index(fields=["project_id"], name="gcpcompsumm_pro_project_id"),
+            models.Index(fields=["invoice_month"], name="gcpcompsumm_pro_invmonth"),
+        ]
+
+    id = models.UUIDField(primary_key=True)
+
+    usage_start = models.DateField(null=False)
+
+    usage_end = models.DateField(null=False)
+
+    instance_type = models.CharField(max_length=50, null=True)
+
+    usage_amount = models.DecimalField(max_digits=24, decimal_places=9, null=True)
+
+    unit = models.CharField(max_length=63, null=True)
+
+    unblended_cost = models.DecimalField(max_digits=24, decimal_places=9, null=True)
+
+    markup_cost = models.DecimalField(max_digits=24, decimal_places=9, null=True)
+
+    currency = models.CharField(max_length=10)
+
+    source_uuid = models.ForeignKey(
+        "api.Provider", on_delete=models.CASCADE, unique=False, null=True, db_column="source_uuid"
+    )
+
+    project_id = models.CharField(unique=False, max_length=256)
+
+    project_name = models.CharField(max_length=256)
+
+    account_id = models.CharField(max_length=50, null=False)
+
+    invoice_month = models.CharField(max_length=256, null=True, blank=True)
+
+    credit_amount = models.DecimalField(max_digits=24, decimal_places=9, null=True, blank=True)
+
+
+class GCPComputeSummaryByServiceP(models.Model):
+    """A summarized partitioned table specifically for UI API queries.
+
+    This table gives a daily breakdown of compute usage by service and instance type.
+
+    """
+
+    class PartitionInfo:
+        partition_type = "RANGE"
+        partition_cols = ["usage_start"]
+
+    class Meta:
+        """Meta for GCPComputeSummaryByServiceP."""
+
+        db_table = "reporting_gcp_compute_summary_by_service_p"
+        indexes = [
+            models.Index(fields=["usage_start"], name="gcpcompsumm_ser_usage_start"),
+            models.Index(fields=["instance_type"], name="gcpcompsumm_ser_insttyp"),
+            models.Index(fields=["invoice_month"], name="gcpcompsumm_ser_invmonth"),
+        ]
```

Comment:
For compute summary by service, should we have an index on the `service_id`?

Reply:
So, I agree with you, but I think we are gonna end up getting rid of these tables later on: https://issues.redhat.com/browse/COST-2106